src/interceptors/xhrInterceptor.js

```diff
@@ -72,7 +72,16 @@ const fakeService = helpers => class XMLHttpRequestInterceptor {
       //TODO: should 'this.response' be the response string or the response json?
       this.responseText = this.response = responseString;
       
-      return successCallback.call(this);
+      const requestDelay = helpers.getDelay();
+      if (!requestDelay) {
+        // for backwards compatibility if there is no requestDelay ot it 0 return callback without setTimeout
+        return successCallback.call(this);
+      } else {
+        return setTimeout(
+          () => successCallback.call(this),
+          requestDelay
+        );
+      }      
```

Comment:
I also wanted to do like this, but in this case we can potentially break something, because with requestDelay=0 previously it called successCallback synchronously (in the same callstack), if we will wrap it with setTimeout(..., 0), it will be called out of callstack on the next event loop tick.
Basically, it shouldn't break anything, but theoretically it is possible. If you feel comfortable with it - version without if is cleaner and ok for me.

I'll change pull request after your approve.
Thanks a lot for the fast response!